install/update.php

```diff
@@ -1299,39 +1442,36 @@ function do_config_migration($migration) {
    /*
     * Table glpi_plugin_fusioninventory_configs
     */
-   $newTable = "glpi_plugin_fusioninventory_configs";
    if ($DB->tableExists('glpi_plugin_tracker_config')) {
       if ($DB->fieldExists('glpi_plugin_tracker_config', 'ssl_only')) {
-         $query = "SELECT * FROM `glpi_plugin_tracker_config`
-            LIMIT 1";
-         $result = $DB->query($query);
-         if ($DB->numrows($result) > 0) {
-            $data = $DB->fetch_assoc($result);
+         $iterator = $DB->request([
+            'FROM'   => 'glpi_plugin_tracker_config',
+            'LIMIT'  => 1
+         ]);
+         if (count($iterator)) {
+            $data = $iterator->next();
             $prepare_Config['ssl_only'] = $data['ssl_only'];
          }
       }
-      //         $query = "SELECT *  FROM `glpi_plugin_tracker_config`
-      //            WHERE `type`='version'
-      //            LIMIT 1, 10";
-      //         $result=$DB->query($query);
-      //         while ($data=$DB->fetch_array($result)) {
-      //            $DB->query("DELETE FROM `glpi_plugin_tracker_config`
-      //               WHERE `ID`='".$data['ID']."'");
-      //         }
    }
    if ($DB->tableExists('glpi_plugin_fusioninventory_configs')) {
       $id = 'id';
       if ($DB->fieldExists('glpi_plugin_fusioninventory_configs', 'ID')) {
          $id = 'ID';
       }
 
-      $query = "SELECT *  FROM `glpi_plugin_fusioninventory_configs`
-         WHERE `type`='version'
-         LIMIT 1, 10";
-      $result=$DB->query($query);
-      while ($data=$DB->fetch_array($result)) {
-         $DB->query("DELETE FROM `glpi_plugin_fusioninventory_configs`
-            WHERE `".$id."`='".$data[$id]."'");
+      $iterator = $DB->request([
+         'FROM'   => 'glpi_plugin_fusioninventory_configs',
+         'WHERE'  => ['type' => 'version'],
+         'START'  => 0,
+         'LIMIT'  => 10
+      ]);
+      while ($data = $iterator->next()) {
+         $DB->delete(
```

Comment:
prepared statement ?
or maybe a single ``delete`` for the two statements ``SELECT`` and ``DELETE``
Actually raw mysql statements supports ``LIMIT`` keyword, did the iterator do ?

Reply:
DBIterator does support LIMIt: http://glpi-developer-documentation.readthedocs.io/en/master/devapi/database/dbiterator.html#request-pager; but not DB::update() nor DB::delete().
I've changed with a prepared statement